pkg/solana/contract.go

```diff
@@ -61,13 +64,53 @@ func NewTracker(spec OCR2Spec, client *Client, transmitter TransmissionSigner, l
 	}
 }
 
+func (c *ContractTracker) Start() error {
+	c.done = make(chan struct{})
+	go c.PollState()
+	return nil
+}
+
+func (c *ContractTracker) PollState() {
+	c.lggr.Debugf("Starting state polling for state: %s, transmissions: %s", c.StateID, c.TransmissionsID)
+	ticker := time.NewTicker(c.client.pollingInterval)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-c.done:
+			c.lggr.Debugf("Stopping state polling for state: %s, transmissions: %s", c.StateID, c.TransmissionsID)
+			return
+		case <-ticker.C:
+			// async poll both transmisison + ocr2 states
+			go func() {
+				ctx, cancel := context.WithTimeout(context.Background(), c.client.contextDuration)
+				defer cancel()
+				if err := c.fetchState(ctx); err != nil {
+					c.lggr.Errorf("error in PollState.fetchState %s", err)
+				}
+			}()
+			go func() {
+				ctx, cancel := context.WithTimeout(context.Background(), c.client.contextDuration)
+				defer cancel()
+				if err := c.fetchLatestTransmission(ctx); err != nil {
+					c.lggr.Errorf("error in PollState.fetchLatestTransmission %s", err)
+				}
+			}()
```

Comment:
Should these goroutines cancel on `c.done`?
Is it OK to proceed without waiting for them to complete?

Reply:
that's a good point, it would be good to cancel the context or wait for it to complete.

not sure how to do this off the top of my mind, but i'll do some digging

Reply:
`package utils` has some relevant helpers, e.g. `ContextFromChan` and `WaitGroupChan`.